### DISCUSSION TITLE
Can't find code that generates web app as seen on librelingo.app site

### DISCUSSION THREAD

**Discussion by Kuhron** [kantord/LibreLingo#3404] [Q&A]
Hello!

I am very interested in contributing to this project! I would use it for language documentation/revitalization projects that I work on, as well as conlanging for fun. I have a lot of Python experience but only moderate web dev / JS, so I'm still learning and trying things out.

I've forked the repo, gotten my local environment set up with all the dependencies, read various tutorials about React and Next, created a small test course, exported and installed that course, and gotten the server running. The home page with course cards loads correctly (although lacking the nice nav bar seen on librelingo.app), and I can click through to enter a course. But then the course home page is blank except for the language name.

I have tried searching all through the repo to find how the course pages are generated on librelingo.app. For instance, where do the strings "Which of these is" and "Type what you hear" on the exercises come from? I cannot find them in the code.

Is the more up-to-date code in a branch other than `main` which is what is being deployed on the app site?

Thanks so much and I really look forward to playing with this app more!

**Comment by Kuhron** [comment]
Okay, I have figured out that librelingo.app is hosted on GitHub Pages from the `gh-pages` branch, so I've pulled that one into my fork and will try to get it running.

**Comment by kantord** [comment]
The app is currently being reworked "from scratch" (ported to another technology), and the current main branch has an early version of the reworked app. The reason is because updating to a later version of Svelte presented too many challenges and a series of problems were blocking further development. I decided that in order to unblock further development of the front-end code I need to move it to a platform that generates fewer challenges with maintenance. Well, until this work is done, further development is still kinda blocked :sweat_smile:  But at least it is not possible to unblock

**Reply by kantord** [reply]
for more context, this is commit has the old version of the frontend code. you should be able to get it running locally if you check out this version: https://github.com/kantord/LibreLingo/tree/e6cb70c8a67134d1e46e3b7ea456c54d0352ec22 albeit it will be difficult to merge your changes to the new version. However if you need it running to test your courses locally, it will still be useful because the rewrite does not touch the "backend" parts (i. e. the coures format)

**Reply by Kuhron** [reply]
What is the new framework? React and Next? I would like to learn some of them in the meantime so that I can be better equipped to help out once it is more ready for contributions.

**Reply by ggadt** [reply]
Ehy there guys, if you are interested, i attempted successfully to update Svelte to v5 and also Sveltekit, despite i don't have so much knowledge of how Svelte works in deep, but at least you could use the project until the new version comes up. If you want to know more, let me know here! 😄
